### PR TITLE
spline #575 Upgrade Spline core to Scala 2.12

### PR DESCRIPTION
**Pull request by wajda**

### CODE REVIEW

File: commons/src/main/scala/za/co/absa/spline/common/future/EstimableFuture.scala [hunk 1]
```diff
@@ -57,7 +61,7 @@ object EstimableFuture {
         new MovingAverageCalculator(10.seconds.toMillis, 0.05)
       })
       val startTime = System.currentTimeMillis
-      future.onSuccess { case _ => durationMeasurer.addMeasurement(System.currentTimeMillis - startTime) }
+      future.foreach { case _ => durationMeasurer.addMeasurement(System.currentTimeMillis - startTime) }
```
Comment (cerveada) [inline commons/src/main/scala/za/co/absa/spline/common/future/EstimableFuture.scala]: What is the reason behind this change?

Comment (wajda) [reply commons/src/main/scala/za/co/absa/spline/common/future/EstimableFuture.scala]: `onSuccess` is deprecated in Scala 2.12

File: persistence/pom.xml [hunk 2]
```diff
@@ -62,9 +62,9 @@
             <version>6.5.0</version>
         </dependency>
         <dependency>
-            <groupId>com.arangodb</groupId>
-            <artifactId>velocypack-module-scala</artifactId>
-            <version>1.1.0</version>
+            <groupId>com.github.wajda.com.arangodb</groupId>
+            <artifactId>velocypack-module-scala_${scala.compat.version}</artifactId>
+            <version>1.1.0.1</version>
```
Comment (cerveada) [inline persistence/pom.xml]: It looks like, they merged your changes. https://github.com/arangodb/java-velocypack-module-scala/releases/tag/v1.2.0

So we can use the original version again, right?

Comment (wajda) [reply persistence/pom.xml]: I'm still waiting for it to appear in Maven Central. It's not there yet - https://repo1.maven.org/maven2/com/arangodb/velocypack-module-scala_2.12/1.2.0/velocypack-module-scala_2.12-1.2.0.pom
